dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry/internal/cdi/OpenTelemetryProducer.java

```diff
@@ -68,41 +68,44 @@ public class OpenTelemetryProducer {
-
-        //Builds tracer provider if user has enabled tracing aspects with config properties
-        if (!checkDisabled(telemetryProperties)) {
-            OpenTelemetry openTelemetry = AccessController.doPrivileged((PrivilegedAction<OpenTelemetry>) () -> {
-                return AutoConfiguredOpenTelemetrySdk.builder()
-                                .addPropertiesCustomizer(x -> telemetryProperties) //Overrides OpenTelemetry's property order
-                                .addResourceCustomizer(this::customizeResource)//Defaults service name to application name
-                                .setServiceClassLoader(Thread.currentThread().getContextClassLoader())
-                                .setResultAsGlobal(false)
-                                .registerShutdownHook(false)
-                                .build()
-                                .getOpenTelemetrySdk();
-            });
-
-            if (openTelemetry != null) {
-                return new OpenTelemetryInfo(true, openTelemetry);
+            HashMap<String, String> telemetryProperties = getTelemetryProperties();
+
+            //Builds tracer provider if user has enabled tracing aspects with config properties
+            if (!checkDisabled(telemetryProperties)) {
+                OpenTelemetry openTelemetry = AccessController.doPrivileged((PrivilegedAction<OpenTelemetry>) () -> {
+                    return AutoConfiguredOpenTelemetrySdk.builder()
+                                    .addPropertiesCustomizer(x -> telemetryProperties) //Overrides OpenTelemetry's property order
+                                    .addResourceCustomizer(this::customizeResource)//Defaults service name to application name
+                                    .setServiceClassLoader(Thread.currentThread().getContextClassLoader())
+                                    .setResultAsGlobal(false)
+                                    .registerShutdownHook(false)
+                                    .build()
+                                    .getOpenTelemetrySdk();
+                });
+
+                if (openTelemetry != null) {
+                    return new OpenTelemetryInfo(true, openTelemetry);
+                }
             }
+            //By default, MicroProfile Telemetry tracing is off.
+            //The absence of an installed SDK is a “no-op” API
+            //Operations on a Tracer, or on Spans have no side effects and do nothing
+            ComponentMetaData cData = ComponentMetaDataAccessorImpl.getComponentMetaDataAccessor().getComponentMetaData();
+            String applicationName = cData.getJ2EEName().getApplication();
+            Tr.info(tc, "CWMOT5100.tracing.is.disabled", applicationName);
+
+            return new OpenTelemetryInfo(false, OpenTelemetry.noop());
+        } catch (Exception e) {
+            Tr.error(tc, Tr.formatMessage(tc, "CEMOT5101.telemetry.error", e));
+            return null;
         }
```

Comment:
We need to return a no-op implementation here, rather than `null` (i.e. the same behaviour as if tracing is disabled).

You're not allowed to return `null` from an `ApplicationScoped` producer method anyway, so if we did return `null` the user would just get a more confusing error message.